go.mod

```diff
@@ -39,7 +39,6 @@ require (
 	go.opentelemetry.io/collector/config/confighttp v0.122.1
 	go.opentelemetry.io/collector/config/configopaque v1.28.1
 	go.opentelemetry.io/collector/config/configretry v1.28.1
-	go.opentelemetry.io/collector/config/configtelemetry v0.122.1
```

Comment:
Why is this removed? Did we forget to remove it in our PRs updating OTE

Reply:
I thought it was missing after `go mod tidy && go mod vendor`, but I got it back